daemon/image.go

```diff
@@ -45,18 +46,25 @@ func (daemon *Daemon) GetImageIDAndPlatform(refOrID string) (image.ID, string, e
 		return "", "", errImageDoesNotExist{ref}
 	}
 
-	for platform := range daemon.stores {
-		if id, err := daemon.stores[platform].referenceStore.Get(namedRef); err == nil {
-			return image.IDFromDigest(id), platform, nil
+	if digest, err := daemon.referenceStore.Get(namedRef); err == nil {
+		// Search the image stores to get the platform, defaulting to host OS.
+		imagePlatform := runtime.GOOS
+		id := image.IDFromDigest(digest)
+		for platform := range daemon.stores {
+			if img, err := daemon.stores[platform].imageStore.Get(id); err == nil {
```

Comment:
@tonistiigi Sure, let me capture that in the document as the follow-up to last week (almost complete and ready to share for discussion - delayed as OOF) and do as a follow-up PR.